sway/desktop/render.c

```diff
@@ -430,14 +481,33 @@ static void render_surface_iterator(struct sway_output *output,
 		wlr_output);
 }
 
+// view will be NULL every time
+static void render_layer_iterator(struct sway_output *output,
+		struct sway_view *view, struct wlr_surface *surface,
+		struct wlr_box *_box, void *_data) {
+	// render the layer's surface
+	render_surface_iterator(output, view, surface, _box, _data);
+
+	struct render_data *data = _data;
+	pixman_region32_t *output_damage = data->damage;
+	struct decoration_data deco_data = data->deco_data;
+
+	// render shadow
+	if (deco_data.shadow && config_should_parameters_shadow()) {
```

Comment:
TODO: Check if layers can have subsurfaces

Reply:
Maybe we could add a pointer to the top most surface to `render_data`? We could add another field to `sway_surface_iterator_func_t` but that would make it more "difficult" to rebase from upstream. Any ideas? :)

Reply:
I have an idea here, will fix later today

Reply:
Can check if it's a subsurface in output.c where we pass the decoration_data. We can just set blur to false there if it's a subsurface!

I'll implement it this evening, but if you're around you're welcome to add it + merge